idaes/core/util/scaling.py

```diff
@@ -230,6 +233,26 @@ def set_scaling_factor(c, v, data_objects=True, overwrite=True):
         # doesn't exist.  This handles the case where you get a constant 0 and
         # need its scale factor to scale the mass balance.
         return 1
+
+    if isinstance(c, ConstraintData):
+        xform_factor = get_constraint_transform_applied_scaling_factor(c)
+        if xform_factor is not None:
+            raise RuntimeError(
+                "Attempted to set constraint scaling factor for transformed constraint. "
+                "Please use only one of set_scaling_factor and constraint_scaling_transform "
+                "per constraint to avoid double scaling."
+            )
+    elif isinstance(c, Constraint):
+        for condata in c.values():
+            xform_factor = get_constraint_transform_applied_scaling_factor(condata)
+            if xform_factor is not None:
+                raise RuntimeError(
+                    "Attempted to set constraint scaling factor for indexed constraint "
+                    "with transformed ConstraintData children. Please use only one of "
```

Comment:
```suggestion
                    f"Attempted to set constraint scaling factor for indexed constraint {c.name}"
                    f"with transformed ConstraintData children. Please use only one of "
```
I think it could be useful to directly point to the constraint where the issue arises, in the case that the user might not be applying scaling to constraints manually, one at a time.